core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -406,6 +398,18 @@ class LogManager(logDirs: Seq[File],
     info(s"Loaded $numTotalLogs logs in ${time.hiResClockMs() - startMs}ms.")
   }
 
+  private def logDirExecutor(logDirAbsolutePath: String, threadNamePrefix: String): ExecutorService = {
+    Executors.newFixedThreadPool(numRecoveryThreadsPerDataDir, new ThreadFactory {
```

Comment:
How about using following style?
```scala
Executors.newFixedThreadPool(numRecoveryThreadsPerDataDir, KafkaThread.nonDaemon(s"$threadNamePrefix-$logDirAbsolutePath", _))
```

Reply:
Yeah, that's much better, thanks! I've inlined the method because it didn't add much.